.github/workflows/ci-build.yml

```diff
@@ -2,71 +2,68 @@ name: CI Build
 
 on:
   - push
+  - pull_request
 
 jobs:
   build:
-    name: Building with Scala ${{ matrix.scala }}
+    name: Building with Scala ${{ matrix.scala-version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        scala: [ 2.12.15, 2.13.6 ]
-
+        include:
+          - { scala-version: 'scala2.12', pk-skip: true }
+          - { scala-version: 'scala2.13', pk-skip: false }
     steps:
-      - name: Checkout the Repository
+      - name: Checkout the repository
         uses: actions/checkout@v2
         with:
           fetch-depth: 0
-
-      - name: Setup Java and Scala with JDK 11
-        uses: olafurpg/setup-scala@v10
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
         with:
+          distribution: 'temurin'
           java-version: 11
-
-      - name: Cache Local SBT Dependencies
+      - name: Cache local Maven repository
         uses: actions/cache@v2
         with:
-          path: |
-            ~/.sbt
-            ~/.ivy2/cache
-            ~/.coursier/cache/v1
-            ~/.cache/coursier/v1
-          key: ${{ runner.os }}-sbt-cache-${{ hashFiles('**/*.sbt') }}-${{ hashFiles('project/build.properties') }}
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
-            ${{ runner.os }}-sbt-cache-
-
-      - name: Check Formatting
-        run: sbt ++${{ matrix.scala }} scalafmtSbtCheck scalafmtCheckAll
-
-      - name: Run CI
-        run: ./scripts/ci.sh
-        env:
-          SCALA_VERSION: ${{ matrix.scala }}
-
-      - name: Upload Coverage Results to Coveralls
-        run: sbt ++${{ matrix.scala }} coveralls
-        env:
-          COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-
-      # This required because of the sonarcloud-github-action docker volume mapping.
-      - name: Prepare for Sonar Cloud Scan
-        if: startsWith(matrix.scala, '2.13')
+            ${{ runner.os }}-maven-
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+      - name: Run scalafix linting
         run: |
-          find . -name scoverage.xml -exec sed -i \
-          's#/home/runner/work/import-export-udf-common-scala/import-export-udf-common-scala#/github/workspace#g' {} +
-
-      - name: Sonar Cloud Scan
-        if: startsWith(matrix.scala, '2.13')
-        uses: sonarsource/sonarcloud-github-action@master
+          mvn --batch-mode clean compile test scalafix:scalafix \
+              -P${{ matrix.scala-version }} \
+              -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn \
+              -DtrimStackTrace=false
+      - name: Run tests and build with Maven
+        run: |
+          mvn --batch-mode clean verify \
```

Comment:
do you need clean here? You clean alreaddy in the previous step...

Reply:
Not really, fixing.